src/DNS_Mgr.cc

```diff
@@ -906,7 +906,7 @@ ValPtr DNS_Mgr::BuildMappingVal(const DNS_MappingPtr& dm) {
     r->Assign(4, h ? std::move(h) : make_intrusive<StringVal>("<none>"));
     r->Assign(5, dm->AddrsSet());
 
-    return r;
+    return std::move(r);
```

Comment:
Do you know if it suggests this also for `std::shared_ptr`. Wouldn't the compiler recognize that it can move?

Reply:
This move is only necessary because this is a type conversion. This is actually `ValPtr{std::move(dm)}`. If the function would return the same type, the compiler would move automatically (guaranteed copy elision) and adding a `move` would actually be a pessimization.

Reply:
> This move is only necessary because this is a type conversion. This is actually ValPtr{std::move(dm)}. If the function would return the same type, the compiler would move automatically (guaranteed copy elision) and adding a move would actually be a pessimization.

Ah that explains that then. I was trying to figure out why exactly the compiler wouldn't automatically elide these. As for the pessimization, a decently-new version of Clang complains when you override the elision so those are easy to catch. I ran into one already where I misread the section that needed to be changed from the Coverity finding.

Reply:
> Unfortunately we really can't change all of these signatures to return the matching types.

I would entertain going through and fixing all of the ones in `detail` namespaces at least though.